lnwallet/fee_estimator.go

```diff
@@ -524,3 +542,205 @@ func (e ExternalFeeSource) ParseResponse(r io.Reader) (map[uint32]uint32, error)
 // A compile-time assertion to ensure that ExternalFeeSource implements the
 // WebAPIFeeSource interface.
 var _ WebAPIFeeSource = (*ExternalFeeSource)(nil)
+
+// WebAPIFeeEstimator is an implementation of the FeeEstimator interface that
+// queries an HTTP-based fee estimation from an existing web API.
+type WebAPIFeeEstimator struct {
+	started uint32 // To be used atomically.
+	stopped uint32 // To be used atomically.
+
+	// apiSource is the backing web API source we'll use for our queries.
+	apiSource WebAPIFeeSource
+
+	// chainNotifier is used by the WebAPIFeeEstimator to determine when to
+	// re-query the API for fresh fees. We re-query whenever a new block
+	// comes in.
+	chainNotifier chainntnfs.ChainNotifier
```

Comment:
Does it need both the notifier and a block epoch attribute? Seems like it could do with one or the other.

Reply:
n/a since we just hit the API on a random timer now